.github/workflows/build.yml

```diff
@@ -0,0 +1,57 @@
+name: build
+
+on: [push, pull_request]
+
+env:
+  GRADLE_OPTS: "-Dorg.gradle.jvmargs=-Xmx4g -Dorg.gradle.daemon=false -Dkotlin.incremental=false"
+
+jobs:
+  jvm:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version:
+          - 1.8
+          - 11
+
+    steps:
+      - name: Cache Gradle dependencies
+        uses: actions/cache@v2
```

Comment:
I'd generally recommend Vlad's excellent native-cache plugin

https://github.com/burrunan/gradle-cache-action

example: https://github.com/open-telemetry/opentelemetry-java/blob/master/.github/workflows/pr-build.yml#L36

Reply:
Want me to just copy/paste this?

Reply:
Well I think the properties / arguments will differ a bit but yeah should mostly be copy-paste :)

Reply:
Why use this? Can we specify paths?

Reply:
The README has a description of why to use it :) Don't need to specify paths since it's Gradle-aware and does it properly automatically

Reply:
So you want me to replace the build "step" and current cache plugin with this?

Reply:
Yeah

Reply:
Done.

Reply:
Seems to have a problem on the CI?
```
Warning: Gradle wrapper configuration is not found at /home/runner/work/gradle-versions-plugin/gradle-versions-plugin/gradle/wrapper/gradle-wrapper.properties.
Will use the current release Gradle version
/usr/bin/git log -1 --quiet --format=%T
fatal: not a git repository (or any of the parent directories): .git
(node:2407) UnhandledPromiseRejectionWarning: Error: The process '/usr/bin/git' failed with exit code 128
    at ExecState._setResult (/home/runner/work/_actions/burrunan/gradle-cache-action/v1.6/dist/cache-action-entrypoint.js:45001:468)
    at ExecState.CheckComplete (/home/runner/work/_actions/burrunan/gradle-cache-action/v1.6/dist/cache-action-entrypoint.js:44994:45)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/burrunan/gradle-cache-action/v1.6/dist/cache-action-entrypoint.js:44961:84)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```